warning: denote infinite loops with `loop { ... }`
  --> $DIR/loop-break-value.rs:26:5
   |
LL |     'while_loop: while true {
   |     ^^^^^^^^^^^^^^^^^^^^^^^ help: use `loop`
   |
   = note: `#[warn(while_true)]` on by default

error[E0571]: `break` with value from a `while` loop
  --> $DIR/loop-break-value.rs:28:9
   |
LL |         break ();
   |         ^^^^^^^^ can only break with a value inside `loop` or breakable block
   |
help: instead, use `break` on its own without a value inside this `while` loop
   |
LL |         break;
   |         ^^^^^

error[E0571]: `break` with value from a `while` loop
  --> $DIR/loop-break-value.rs:30:13
   |
LL |             break 'while_loop 123;
   |             ^^^^^^^^^^^^^^^^^^^^^ can only break with a value inside `loop` or breakable block
   |
help: instead, use `break` on its own without a value inside this `while` loop
   |
LL |             break;
   |             ^^^^^

error[E0571]: `break` with value from a `while let` loop
  --> $DIR/loop-break-value.rs:38:12
   |
LL |         if break () {
   |            ^^^^^^^^ can only break with a value inside `loop` or breakable block
   |
help: instead, use `break` on its own without a value inside this `while let` loop
   |
LL |         if break {
   |            ^^^^^

error[E0571]: `break` with value from a `while let` loop
  --> $DIR/loop-break-value.rs:43:9
   |
LL |         break None;
   |         ^^^^^^^^^^ can only break with a value inside `loop` or breakable block
   |
help: instead, use `break` on its own without a value inside this `while let` loop
   |
LL |         break;
   |         ^^^^^

error[E0571]: `break` with value from a `while let` loop
  --> $DIR/loop-break-value.rs:49:13
   |
LL |             break 'while_let_loop "nope";
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can only break with a value inside `loop` or breakable block
   |
help: instead, use `break` on its own without a value inside this `while let` loop
   |
LL |             break;
   |             ^^^^^

error[E0571]: `break` with value from a `for` loop
  --> $DIR/loop-break-value.rs:56:9
   |
LL |         break ();
   |         ^^^^^^^^ can only break with a value inside `loop` or breakable block
   |
help: instead, use `break` on its own without a value inside this `for` loop
   |
LL |         break;
   |         ^^^^^

error[E0571]: `break` with value from a `for` loop
  --> $DIR/loop-break-value.rs:57:9
   |
LL |         break [()];
   |         ^^^^^^^^^^ can only break with a value inside `loop` or breakable block
   |
help: instead, use `break` on its own without a value inside this `for` loop
   |
LL |         break;
   |         ^^^^^

error[E0571]: `break` with value from a `for` loop
  --> $DIR/loop-break-value.rs:64:13
   |
LL |             break 'for_loop Some(17);
   |             ^^^^^^^^^^^^^^^^^^^^^^^^ can only break with a value inside `loop` or breakable block
   |
help: instead, use `break` on its own without a value inside this `for` loop
   |
LL |             break;
   |             ^^^^^

error[E0308]: mismatched types
  --> $DIR/loop-break-value.rs:4:31
   |
LL |     let val: ! = loop { break break; };
   |                               ^^^^^ expected !, found ()
   |
   = note: expected type `!`
              found type `()`

error[E0308]: mismatched types
  --> $DIR/loop-break-value.rs:11:19
   |
LL |             break 123;
   |                   ^^^ expected &str, found integer
   |
   = note: expected type `&str`
              found type `{integer}`

error[E0308]: mismatched types
  --> $DIR/loop-break-value.rs:16:15
   |
LL |         break "asdf";
   |               ^^^^^^ expected i32, found reference
   |
   = note: expected type `i32`
              found type `&'static str`

error[E0308]: mismatched types
  --> $DIR/loop-break-value.rs:21:31
   |
LL |             break 'outer_loop "nope";
   |                               ^^^^^^ expected i32, found reference
   |
   = note: expected type `i32`
              found type `&'static str`

error[E0308]: mismatched types
  --> $DIR/loop-break-value.rs:73:26
   |
LL |                 break 'c 123;
   |                          ^^^ expected (), found integer
   |
   = note: expected type `()`
              found type `{integer}`

error[E0308]: mismatched types
  --> $DIR/loop-break-value.rs:80:15
   |
LL |         break (break, break);
   |               ^^^^^^^^^^^^^^ expected (), found tuple
   |
   = note: expected type `()`
              found type `(!, !)`

error[E0308]: mismatched types
  --> $DIR/loop-break-value.rs:85:15
   |
LL |         break 2;
   |               ^ expected (), found integer
   |
   = note: expected type `()`
              found type `{integer}`

error[E0308]: mismatched types
  --> $DIR/loop-break-value.rs:90:9
   |
LL |         break;
   |         ^^^^^
   |         |
   |         expected integer, found ()
   |         help: give it a value of the expected type: `break value`
   |
   = note: expected type `{integer}`
              found type `()`

error: aborting due to 16 previous errors

Some errors have detailed explanations: E0308, E0571.
For more information about an error, try `rustc --explain E0308`.
